File: AssetManager/Helpers/NetworkInfo.vb
Namespace NetworkInfo

    Public Module NetInfo
        Private _currentDomain As String = "core.co.fairfield.oh.us"
        Public ReadOnly Property CurrentDomain As String
            Get
                Return _currentDomain
            End Get
        End Property

        Private DomainNames As New Dictionary(Of Databases, String)() From
        {
            {Databases.test_db, "core.co.fairfield.oh.us"},
            {Databases.asset_manager, "core.co.fairfield.oh.us"},
            {Databases.vintondd, "vinton.local"}
        }

        Private SubnetLocations As New Dictionary(Of String, String)() From
        {
            {"10.10.80.0", "Admin"},
            {"10.10.81.0", "OC"},
            {"10.10.82.0", "DiscoverU"},
            {"10.10.83.0", "FRS"},
            {"10.10.84.0", "PRO"},
            {"10.10.85.0", "Art & Clay"}
        }

        Public Function LocationOfIP(ip As String) As String
            Dim Subnet = ip.Substring(0, 8) & ".0"
            If SubnetLocations.ContainsKey(Subnet) Then
                Return SubnetLocations(Subnet)
            Else
                Return String.Empty
            End If
        End Function

        Public Function SetCurrentDomain(database As Databases) As String
            _currentDomain = DomainNames(database)
            SecurityTools.ClearAdminCreds()
            If database = Databases.vintondd Then
                SecurityTools.VerifyAdminCreds("Credentials for Vinton AD")
            End If
            Return DomainNames(database)
        End Function

    End Module
End Namespace

File: AssetManager.ConvertedToC#/Tools/NetworkInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AssetManager
{
    public static class NetworkInfo
    {
        private static string _currentDomain = "core.co.fairfield.oh.us";
        public static string CurrentDomain
        {
            get { return _currentDomain; }
        }

        private static Dictionary<Databases, string> DomainNames = new Dictionary<Databases, string>
        {
            {Databases.test_db, "core.co.fairfield.oh.us"},
            {Databases.asset_manager, "core.co.fairfield.oh.us"},
            {Databases.vintondd, "vinton.local"}
        };

        private static Dictionary<string, string> SubnetLocations = new Dictionary<string, string>
        {
            {"10.10.80.0", "Admin"},
            {"10.10.81.0", "OC"},
            {"10.10.82.0", "DiscoverU"},
            {"10.10.83.0", "FRS"},
            {"10.10.84.0", "PRO"},
            {"10.10.85.0", "Art & Clay"}
        };

        public static string LocationOfIP(string ip)
        {
        
            try
            {
                string Subnet = ip.Substring(0, 8) + ".0";
                if (SubnetLocations.ContainsKey(Subnet))
                {
                    
                    return SubnetLocations[Subnet];
                }
                else
                {
                    return string.Empty;
                }
            }
            catch
            {
                return string.Empty;
            }
        }

        public static string SetCurrentDomain(Databases database)
        {
            _currentDomain = DomainNames[database];
            SecurityTools.ClearAdminCreds();
            if (database == Databases.vintondd)
            {
                SecurityTools.VerifyAdminCreds("Credentials for Vinton AD");
            }
            return DomainNames(database);
        }


    }
}
